lightning/lightning.go

```diff
@@ -284,6 +312,11 @@ func (l *Lightning) run(taskCfg *config.Config) (err error) {
 }
 
 func (l *Lightning) Stop() {
+	l.cancelLock.Lock()
```

Comment:
This explicit cancel is not needed? If lightning run in server mod, the cancel task action will call this cancel? If lightning run in once mod, `l.shutdown` will stop lightning

Reply:
A test expects this behaviour, that when lightning shutdown task should be canceled to avoid futhur actions (it's remove checkpoint in fact)

https://github.com/pingcap/tidb-lightning/blob/ccf7f3c3e5a80875a3a21b9b02e2657e7748a560/tests/checkpoint_chunks/run.sh#L76-L78